src/Model/ModelManager.php

```diff
@@ -438,39 +464,17 @@ public function modelReverseTransform($class, array $array = [])
         $instance = $this->getModelInstance($class);
         $metadata = $this->getMetadata($class);
 
-        $reflClass = $metadata->reflClass;
         foreach ($array as $name => $value) {
-            $reflection_property = false;
             // property or association ?
             if (\array_key_exists($name, $metadata->fieldMappings)) {
                 $property = $metadata->fieldMappings[$name]['fieldName'];
-                $reflection_property = $metadata->reflFields[$name];
             } elseif (\array_key_exists($name, $metadata->associationMappings)) {
                 $property = $metadata->associationMappings[$name]['fieldName'];
             } else {
                 $property = $name;
             }
 
-            $setter = 'set'.$this->camelize($name);
-
-            if ($reflClass->hasMethod($setter)) {
-                if (!$reflClass->getMethod($setter)->isPublic()) {
-                    throw new PropertyAccessDeniedException(sprintf('Method "%s()" is not public in class "%s"', $setter, $reflClass->getName()));
-                }
-
-                $instance->$setter($value);
-            } elseif ($reflClass->hasMethod('__set')) {
-                // needed to support magic method __set
-                $instance->$property = $value;
-            } elseif ($reflClass->hasProperty($property)) {
-                if (!$reflClass->getProperty($property)->isPublic()) {
-                    throw new PropertyAccessDeniedException(sprintf('Property "%s" is not public in class "%s". Maybe you should create the method "set%s()"?', $property, $reflClass->getName(), ucfirst($property)));
-                }
-
-                $instance->$property = $value;
-            } elseif ($reflection_property) {
-                $reflection_property->setValue($instance, $value);
-            }
+            $this->propertyAccessor->setValue($instance, $property, $value);
```

Comment:
I was thinking about wrapping this with `try` and `catch` to instead of throwing `NoSuchPropertyException`, throwing other one... but I always struggle with the right exception... so if there is any suggestion 😬

Reply:
I would love to help but I'm not sure what you mean… is your sentence maybe missing a word? And why would you wrap the exception? Does it need to be caught? Or is there useful information that needs to be added?

Reply:
sorry! I also struggle with the language 😛 
Yeah, just to clarify I meant something like:

```php
try {
    $this->propertyAccessor->setValue($instance, $property, $value);
} catch (NoSuchPropertyException $exception) {
    throw new \BadMethodCallException('...');
}
```
It doesn't need to be caught the Symfony one and there is no useful information, but I thought it would be better to have control of what Exception are we throwing, so if this is needed to be changed in the future (unlikely to happen) we don't depend on the Symfony one.

Reply:
IMO the Symfony exception in this case is good enough and represents what is happening. Unless we add more value with our exception I would leave it without try catch

Reply:
If we need to change it in the future, and it's not meant to be caught, we will not see that as a BC-break.

Reply:
Should we still use try catch for the 3.x branch since we're currently changing the exception thrown ?

Reply:
Ok, perfect then!

> Should we still use try catch for the 3.x branch since we're currently changing the exception thrown ?

The current exception thrown doesn't exist anymore, so if someone reaches that case it would fail.